payjoin/src/receive/v2/mod.rs

```diff
@@ -60,6 +61,36 @@ pub struct Receiver {
     context: SessionContext,
 }
 
+/// A wrapper around the receiver session. The receiver session is accessible only after it has been persisted via `persist`.
+pub struct EphemeralReceiver {
+    inner: Receiver,
+}
+
+impl EphemeralReceiver {
+    pub fn new(
+        address: Address,
+        directory: impl IntoUrl,
+        ohttp_keys: OhttpKeys,
+        expire_after: Option<Duration>,
+    ) -> Result<EphemeralReceiver, IntoUrlError> {
+        let receiver = Receiver::new(address, directory, ohttp_keys, expire_after)?;
+        Ok(EphemeralReceiver { inner: receiver })
```

Comment:
I prefer inline definition to assignment here `inner: Receiver::new(...)?`

Reply:
Done